src/history/hs.c

```diff
@@ -750,9 +750,18 @@ __wt_hs_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_RECONCILE *r, WT_M
                         WT_ERR(__hs_insert_record(session, cursor, btree_id, key, upd,
                           WT_UPDATE_MODIFY, modify_value, stop_ts_pair));
                         __wt_scr_free(session, &modify_value);
-                    } else
+                    } else {
                         WT_ERR(__hs_insert_record(session, cursor, btree_id, key, upd,
                           WT_UPDATE_STANDARD, full_value, stop_ts_pair));
+                        /*
+                         * If we are evicting, we can now free older updates which have already been
+                         * written to the history store. However, we can only free the updates after
+                         * an original full update on the chain being inserted into the history
+                         * store also as a full update.
```

Comment:
The wording here doesn't really make sense, perhaps you should say:
However we can only free updates after an original full update on the update chain.

Reply:
I want to be more specific. The full update must be inserted into history store as a full update. If we plan to insert it as a reverse modify in the future, we cannot free here.

Reply:
Then how aboutL
However we can only free updates after an original full update is inserted as a full update.